lib/cpp/src/thrift/concurrency/PlatformThreadFactory.h

```diff
@@ -22,9 +22,7 @@
 
 // clang-format off
 #include <thrift/thrift-config.h>
-#if USE_BOOST_THREAD
-#  include <thrift/concurrency/BoostThreadFactory.h>
-#elif USE_STD_THREAD
+#if USE_STD_THREAD
```

Comment:
I'd really recommend getting rid of PosixThreadFactory.h and move the contents of StdThreadFactory.h into PlatformThreadFactory.h, and maybe rename it to TThreadFactory using std::thread only.

Reply:
> 
> 
> I'd really recommend getting rid of PosixThreadFactory.h and move the contents of StdThreadFactory.h into PlatformThreadFactory.h, and maybe rename it to TThreadFactory using std::thread only.

Posixthread has policy and priority settings which are lack in std::thread,So deleting it may break many existing codes,and these settings may be important for performance.
Of course I can port these settings to std::thread if native_handle() is pthread,but I would like to change in another PR.

Reply:
Tweaking thread priority is usually a slippery slope... and the pthread code has platform incompatibilities.  We should remove it all if we're going to do this. :)